### ISSUE TITLE
Inputs with no content to parse are considered valid

### ISSUE THREAD

**Issue by polarathene** [lycheeverse/lychee#665]
With remote inputs, if the URI is to a web server and the page does not exist, it may respond with (or redirect to) a 404 page.
- In some cases, you may want to use `lychee` to link check the content of a custom 404 page itself, but that should probably be opt-in?
- In my case, the server would respond with 404 status, but no additional content (response header `content-length: 0`).
  - This was a broken input link, but `lychee` received a response, and presumably considered it valid, despite no content to parse?

---

Is this expected behaviour for `lychee`? When does it consider the input invalid?

Is it only a failure for inputs when:
- The request times out
- DNS lookup fails

A response with `content-length: 0` might be valid for an actual link being checked (while the 404 status may not), but for inputs at least, no content seems like a bad input? (_unless it is dynamic, like a queue_)

---

It might be useful to log a warning about such inputs. Or possibly fail when no inputs have any content? The only hint of such presently would be that no links were found.

**Comment by mre** [comment]
Maybe we should print a warning if an input does not have any content. It might require some refactoring, though.
For instance I'm fine if a URL doesn't have any content in case it gets detected recursively (once #78 finally lands...).
However for inputs, URLs without any content are definitely suspicious.

I don't really have a strong opinion on this. Would like to hear other people's thoughts before we make a change.

**Comment by mre** [comment]
Since we haven't received any additional feedback or strong opinions on this in the past 2.5 years, I'll close this issue for now. If anyone encounters problems with empty inputs in practice, please feel free to open a new issue with specific examples and use cases. Thanks @polarathene for raising this consideration!

**Comment by mre** [comment]
Also, I just realized this is a partial duplicate of https://github.com/lycheeverse/lychee/issues/918. Since there's a second issue, maybe we should look at it again. Let's move the discussion to the other issue as it's older.
